urllib3/poolmanager.py

```diff
@@ -301,8 +302,12 @@ def urlopen(self, method, url, redirect=True, **kw):
         u = parse_url(url)
         conn = self.connection_from_host(u.host, port=u.port, scheme=u.scheme)
 
-        kw['assert_same_host'] = False
-        kw['redirect'] = False
+        # Rewind body position, if needed. Record current position
```

Comment:
The PoolManager.request method docstring says something about "custom cross-host redirect logic"; this is now the only redirect logic.